File: budget-management-backend/src/main/java/com/suika/bm/model/product/Expense.java
package com.suika.bm.model.product;

import com.suika.bm.model.enums.ExpenseCategory;
import com.suika.bm.model.enums.PaymentMethod;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.ToString;

import java.time.LocalDate;

@Getter
@Setter
@NoArgsConstructor
@ToString
public class Expense {

    Long id;

    String label;

    float amount;

    LocalDate date;

    ExpenseCategory category;

    PaymentMethod paymentMethod;

    boolean hide;

    Long relatedBudgetItemId;

    boolean autoAdded;
}

File: budget-management-frontend/src/app/model/interfaces/expense.ts
import { ExpenseCategory } from "../enums/expenseCategory";
import { PaymentMethod } from "../enums/PaymentMethod";

export interface Expense {
    id?: number;
    label: string;
    amount: number;
    date?: Date;
    category?: ExpenseCategory;
    paymentMethod?: PaymentMethod;
    selected?: boolean;
    hide?: boolean;
}
